editor/editor_file_system.cpp

```diff
@@ -2746,6 +2746,30 @@ Error EditorFileSystem::_reimport_file(const String &p_file, const HashMap<Strin
 		ERR_FAIL_COND_V_MSG(!found, ERR_FILE_NOT_FOUND, vformat("Can't find file '%s' during file reimport.", p_file));
 	}
 
+	// Perform checks in advance for contents that will never be valid resources,
+	// and give user-friendly error messages.
+	ERR_FAIL_COND_V_MSG(FileAccess::get_size(p_file) == 0, ERR_FILE_CORRUPT,
+			vformat("Resource file is corrupt (empty size): %s", p_file));
+
+	Error fa_err;
+	Ref<FileAccess> file = FileAccess::open(p_file, FileAccess::READ, &fa_err);
+	ERR_FAIL_COND_V_MSG(fa_err != OK, fa_err,
+			vformat("Can't open resource file: %s", p_file));
+
+	// Handle Git LFS pointer files that haven't been replaced
+	// with the actual file by running `git lfs pull` yet.
+	// Check the first bytes to make sure we aren't trying to read a binary file.
+	// Otherwise, `get_line()` will print Unicode errors.
+	const uint8_t first_byte = file->get_8();
+	const uint8_t second_byte = file->get_8();
+	const uint8_t third_byte = file->get_8();
+	const uint8_t fourth_byte = file->get_8();
+	if (first_byte == 'v' && second_byte == 'e' && third_byte == 'r' && fourth_byte == 's') { // "vers" in "version"
```

Comment:
There's a similar check in place in the OBJ importer, although it's designed to check if the file is binary instead of being text: https://github.com/godotengine/godot/blob/master/editor/import/3d/resource_importer_obj.cpp#L204

It's likely possible to write this in a more robust/efficient manner, but it probably works well enough as-is.

Reply:
It's a simple ASCII string, so it might be better to just check it in the loop instead of using `get_line(), which would still print errors if file start with `vers` and have invalid UTF-8 sequences.

```c++
	const char *sample = "version https://git-lfs";
	bool match = true;
	for (int i = 0; sample[i] != 0; i++) {
		if (f->eof_reached() || sample[i] != f->get_8()) {
			match = false;
			break;
		}
	}
	ERR_FAIL_COND_V_MSG(match, ERR_FILE_UNRECOGNIZED,
				vformat("Resource file is a Git LFS pointer (run `git lfs pull` to fetch it): %s", p_file));
```